rust/src/data/ffi.rs

```diff
@@ -403,6 +403,35 @@ pub extern "C" fn opendp_data__to_string(this: *const AnyObject) -> FfiResult<*m
         FfiResult::Ok)
 }
 
+/// wrap an AnyObject in an FfiResult::Ok(this)
+#[no_mangle]
+pub extern "C" fn ffiresult_ok(this: *const AnyObject) -> *const FfiResult<*const AnyObject> {
+    util::into_raw(FfiResult::Ok(this))
+}
```

Comment:
Why have a function for this? `FfiResult` is a C union, Python can deal with it fine. It's intended to be returned on the stack, not as a pointer, so Python doesn't have to deal with memory management for return values. (I think these functions leak in the callback code.)

Reply:
This was my first approach, but I can't seem to escape the python refcounter. The C union needs to outlive the python stack frame it was defined in. Even storing the payload and ffiresult in globals is not enough. I ended up using this to ensure that Rust has ownership and can free it.

Reply:
Ah right, you can't get Python to give you stack value in this situation.